WhateverGreen/kern_igfx.cpp

```diff
@@ -1058,47 +1058,55 @@ bool IGFX::applySklAsKblPatches(IOService *that) {
 	DBGLOG("igfx", "disabling VP9 hw decode support on Skylake with KBL kexts");
 	that->removeProperty("IOGVAXDecode");
 
-	auto hevcDecodeCap = OSDynamicCast(OSDictionary, that->getProperty("IOGVAHEVCDecodeCapabilities"));
-	if (!hevcDecodeCap)
-		return false;
+	// SKL does not support 10-bit hardware encoding/decoding, and this causes freezing when attempting to do so.
+	// The removal of profile 2 under VTSupportedProfileArray allows fallback to software encoding/decoding.
+	// Thanks dhinakg and aben for finding this.
+	const char *hevcCapProps[] = { "IOGVAHEVCDecodeCapabilities", "IOGVAHEVCEncodeCapabilities" };
+	for (auto prop : hevcCapProps) {
+		auto hevcCap = OSDynamicCast(OSDictionary, that->getProperty(prop));
+		if (!hevcCap)
+			return false;
 
-	auto newHevcDecodeCap = OSDictionary::withDictionary(hevcDecodeCap);
-	if (!newHevcDecodeCap)
-		return false;
+		auto newHevcCap = OSDictionary::withDictionary(hevcCap);
+		if (!newHevcCap)
+			return false;
 
-	auto vtSuppProf = OSDynamicCast(OSArray, newHevcDecodeCap->getObject("VTSupportedProfileArray"));
-	if (!vtSuppProf) {
-		newHevcDecodeCap->release();
-		return false;
-	}
+		auto vtSuppProf = OSDynamicCast(OSArray, newHevcCap->getObject("VTSupportedProfileArray"));
+		if (!vtSuppProf) {
+			newHevcCap->release();
+			return false;
+		}
 
-	auto newVtSuppProf = OSArray::withArray(vtSuppProf);
-	if (!newVtSuppProf) {
-		newHevcDecodeCap->release();
-		return false;
-	}
+		auto newVtSuppProf = OSArray::withArray(vtSuppProf);
+		if (!newVtSuppProf) {
+			newHevcCap->release();
+			return false;
+		}
 
-	unsigned int count = newVtSuppProf->getCount();
-	bool found = false;
-	for (unsigned int i = 0; i < count; i++) {
-		auto num = OSDynamicCast(OSNumber, newVtSuppProf->getObject(i));
-		if (!num) break;
+		unsigned int count = newVtSuppProf->getCount();
+		for (unsigned int i = 0; i < count; i++) {
+			auto num = OSDynamicCast(OSNumber, newVtSuppProf->getObject(i));
+			if (!num) {
+				newVtSuppProf->release();
+				newHevcCap->release();
+				return false;
+			}
 
-		if (num->unsigned8BitValue() == 2) {
-			DBGLOG("igfx", "removing profile 2 from VTSupportedProfileArray/IOGVAHEVCDecodeCapabilities index %u on Skylake with KBL kexts", i);
-			newVtSuppProf->removeObject(i);
-			found = true;
-			break;
+			if (num->unsigned8BitValue() == 2) {
+				DBGLOG("igfx", "removing profile 2 from VTSupportedProfileArray/%s index %u on Skylake with KBL kexts", prop, i);
+				newVtSuppProf->removeObject(i);
+				break;
+			}
 		}
-	}
 
-	newHevcDecodeCap->setObject("VTSupportedProfileArray", newVtSuppProf);
-	newVtSuppProf->release();
+		newHevcCap->setObject("VTSupportedProfileArray", newVtSuppProf);
+		newVtSuppProf->release();
 
-	that->setProperty("IOGVAHEVCDecodeCapabilities", newHevcDecodeCap);
-	newHevcDecodeCap->release();
+		that->setProperty(prop, newHevcCap);
+		newHevcCap->release();
+	}
 
-	return found;
+	return true;
```

Comment:
Needs a var set to true in the end of the iteration in my opinion.

Reply:
Above. So if everything flows, nothing wrong happens, hence true.